src/top-bar/period-select/period-menu.js

```diff
@@ -1,18 +1,32 @@
 import { Menu, MenuItem } from '@dhis2/ui'
+import moment from 'moment'
 import PropTypes from 'prop-types'
 import React from 'react'
+import { useAppContext } from '../../app-context/index.js'
 import { createHref } from '../../navigation/index.js'
 import { useSelectionContext } from '../../selection-context/index.js'
 import { getFixedPeriodsByTypeAndYear } from '../../shared/index.js'
 import classes from './period-menu.module.css'
 
+const formatPeriod = ({ period, periodType, dateFormat }) => {
+    if (periodType === 'Daily') {
+        return moment(period.displayName, 'YYYY-MM-DD').format(
+            // moment format tokens are case sensitive
+            // see https://momentjs.com/docs/#/parsing/string-format/
+            dateFormat === 'yyyy-mm-dd' ? 'YYYY-MM-DD' : 'DD-MM-YYYY'
+        )
+    }
+    return period.displayName
+}
+
 const PeriodMenu = ({ periodType, year }) => {
     const {
         workflow,
         period: selectedPeriod,
         orgUnit,
         selectPeriod,
     } = useSelectionContext()
+    const { dateFormat } = useAppContext()
     const periods = getFixedPeriodsByTypeAndYear(periodType, year, {
         reversePeriods: true,
     })
```

Comment:
I guess, if we go with what I'm proposing, it would make sense to change the signature for this function, maybe sth like this:
```js
getFixedPeriodsByTypeAndYear({
    periodType, 
    year,
    formatYyyyMmDd,
    filterFuturePeriods,
    config: {
        reversePeriods: true,
    },
})
```

Reply:
I did intend to implement this in the fixed-periods module initially but thought the changes would be complex relative to the fix in the period menu component.

I'll try implementing your proposal and see what the code ends up looking like.